File: app/src/main/java/ru/nikshlykov/englishwordsapp/LinkDao.java
package ru.nikshlykov.englishwordsapp;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface LinkDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    long insert(Link link);

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    List<Long> insertMultiple(List<Link> links);

    @Delete
    int delete(Link link);

    @Delete
    int deleteMultiple(List<Link> links);

    @Query("SELECT * FROM Links WHERE WordId = :wordId")
    Link[] getLinksByWordId(long wordId);
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/daos/LinkDao.kt
package ru.nikshlykov.englishwordsapp.db.daos

import androidx.room.*
import ru.nikshlykov.englishwordsapp.db.models.Link

@Dao
interface LinkDao {

  // TODO посмотреть, как Room отдаёт этот Long.
  //  Сейчас из БД возвращается каждый раз больший на один, но key в entity нет.
  @Insert(onConflict = OnConflictStrategy.IGNORE)
  suspend fun insert(link: Link): Long

  @Delete
  suspend fun delete(link: Link): Int

  @Query("SELECT * FROM Links WHERE WordId = :wordId")
  suspend fun getLinksByWordId(wordId: Long): List<Link>
}